pkg/distributor/distributor.go

```diff
@@ -1115,15 +1147,23 @@ func (d *Distributor) metricsMiddleware(next push.Func) push.Func {
 
 		numSamples := 0
 		numExemplars := 0
+		numHistograms := 0
 		for _, ts := range req.Timeseries {
 			numSamples += len(ts.Samples)
 			numExemplars += len(ts.Exemplars)
+			numHistograms += len(ts.Histograms)
 		}
 
 		d.incomingRequests.WithLabelValues(userID).Inc()
-		d.incomingSamples.WithLabelValues(userID).Add(float64(numSamples))
-		d.incomingExemplars.WithLabelValues(userID).Add(float64(numExemplars))
-		d.incomingMetadata.WithLabelValues(userID).Add(float64(len(req.Metadata)))
+		if numSamples+numHistograms > 0 {
+			d.incomingSamples.WithLabelValues(userID).Add(float64(numSamples + numHistograms))
+		}
+		if numExemplars > 0 {
+			d.incomingExemplars.WithLabelValues(userID).Add(float64(numExemplars))
+		}
+		if len(req.Metadata) > 0 {
+			d.incomingMetadata.WithLabelValues(userID).Add(float64(len(req.Metadata)))
+		}
```

Comment:
Same comment on optimizations and benchmarks. Otherwise why make code more complicated?

Reply:
fixed